### PR TITLE
automate in app batch changes changelog constants, refactor release sequence to consolidate baking steps

### PR DESCRIPTION
**Pull request by coury-clark**
Closes https://github.com/sourcegraph/sourcegraph/issues/48631

Automate the manual constant upgrade for batch changes in app changelog. This also does a bit of a refactor so that we can consolidate all the baking steps in the release process into a single command.

## Test plan

I ran the command in dry run mode for both, but only tested against the batch changes since it's quite tedious to set up a test scenario for older versions of MVU.

I ran `pnpm run release release:bake-content` with 5.0.0 set as the active version:

CleanShot 2023-03-03 at 14 34 30@2x

## App preview:

- [Web](https://sg-web-cclark-automate-batch-changes.onrender.com/search)

Check out the [client app preview documentation](https://docs.sourcegraph.com/dev/how-to/client_pr_previews) to learn more.

### CODE REVIEW

File: dev/release/src/release.ts [hunk 1]
```diff
@@ -988,45 +990,73 @@ ${patchRequestIssues.map(issue => `* #${issue.number}`).join('\n')}`
...
+            const mainBranchEdits: Edit[] = [
+                ...multiVersionSteps,
+                ...srcCliSteps,
+                ...batchChangesInAppChangelog(new SemVer(release.version.version).inc('minor'), false), // in the next main branch this will reflect the guessed next version
+            ]
+
+            const releaseBranchEdits: Edit[] = [
+                ...multiVersionSteps,
+                ...srcCliSteps,
+                ...batchChangesInAppChangelog(release.version, true),
```
Comment (coury-clark) [inline dev/release/src/release.ts:1028]: is this accurate (always assuming we show the changelog in the release)? or should we just default to whatever value is there?

Comment (coury-clark) [reply dev/release/src/release.ts:1029]: @eseliger bump!

Comment (eseliger) [reply dev/release/src/release.ts:1029]: Sorry!

The idea is:
- Before the cut: On main, bump the version to the release version we're about to cut
- Cherry pick that commit into the release branch
- After the cut on main: Reset the var to false always (don't cherry pick this one)

So on the release branch, the flag should never be adjusted, only the version :)

Comment (coury-clark) [reply dev/release/src/release.ts]: Okay, updated such that we will only overwite the visible field on `main`, and the release branch will retain the value that was already present.

Example PRs:

`main` -> https://github.com/sourcegraph/sourcegraph/pull/48771
`release branch` -> https://github.com/sourcegraph/sourcegraph/pull/48772

Comment (eseliger) [reply dev/release/src/release.ts]: Assuming #48771 would've reset the bool flag to false if it wasn't already, and the release branch left it untouched, that looks correct https://github.com/sourcegraph/sourcegraph/pull/48771/files#diff-d7954eae0e0a8bd5036c0f015540bca13858a76518a5bb5bfd6a63759e32cb4dR22

Will look through the code again now

**Review by eseliger — APPROVED** [review]
looks good, thanks for helping us do less manual labor in the future!!

**Pull request merged**